modules/versioning.nix

```diff
@@ -0,0 +1,59 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  version = config.nix-bitcoin.configVersion;
+
+  # Sorted by increasing version numbers
+  changes = [
+    # None yet
+    # {
+    #   version = "0.1";
+    #   condition = config.services.foo.enabled;
+    #   message = ''
+    #     demo message
+    #   '';
+    # }
+  ];
+
+  incompatibleChanges = optionals
+    (version != null && versionOlder lastChange)
+    (builtins.filter (change: versionOlder change && (change.condition or true)) changes);
```

Comment:
what does this "or true" do?

Reply:
It returns `true` if the change has no `condition` attribute.

Reply:
I see. Hadn't noticed that construction before.